### PR TITLE
Addition of two examples to SchedulingIntegrationTests to illustrate …

### LINKED ISSUE #994: Define a goal type that instantiates one single activity
### Checked for duplicates

Yes - I've already checked

### Alternatives considered

Yes - and alternatives don't suffice

### Related problems

_No response_

### Describe the feature request

I want to add the capability to express a goal that results in the instantiation of one single activity. 
The use should be able to add logic or temporal conditions to the goal such as indicating the specific interval at where the activity should be place (for example a direct to earth communication from the rover at the end of a sol)

### PR DESCRIPTION
**Pull request by jmdelfa**
…how to model a unitary goal.

* **Tickets addressed:** Closes https://github.com/NASA-AMMOS/aerie/issues/994
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)

### CODE REVIEW

**Comment by adrienmaillard** [comment]
@jmdelfa I was reading these tests and I felt bad for our users having to do all the boilerplate with `Windows` and `Interval` so I made another commit to your branch adding the possibility for the user to give `Interval` or `Temporal.Instant` directly to `CoexistenceGoal` and then we do the conversion ourselves. I have also modified the examples you gave.

**Comment by adrienmaillard** [comment]
The only downside for you is that you would have to write more documentation to show these use-cases to the user :)

**Comment by jmdelfa** [comment]
I agree with the modifications proposed by @adrienmaillard

File: scheduler-worker/scheduling-dsl-compiler/src/libs/scheduler-edsl-fluent-api.ts [hunk 1]
```diff
@@ -393,7 +393,16 @@ export class Goal {
...
 
-    if (opts.forEach instanceof WindowsEDSL.Windows) {
+    let localForEach;
+    if(opts.forEach instanceof WindowsEDSL.Interval){
+      localForEach = Windows.Value(true, opts.forEach).assignGaps(false);
+    } else if(opts.forEach instanceof Temporal.Instant){
+      localForEach = Windows.Value(true,Interval.At(opts.forEach)).assignGaps(false);
+    }else{
+      localForEach = opts.forEach;
+    }
```
Comment (mattdailis) [inline scheduler-worker/scheduling-dsl-compiler/src/libs/scheduler-edsl-fluent-api.ts]: Here are the spaces you and I discussed adding - it doesn't look to me like they made it into the final version

```suggestion
    if (opts.forEach instanceof WindowsEDSL.Interval) {
      localForEach = Windows.Value(true, opts.forEach).assignGaps(false);
    } else if (opts.forEach instanceof Temporal.Instant) {
      localForEach = Windows.Value(true,Interval.At(opts.forEach)).assignGaps(false);
    } else {
      localForEach = opts.forEach;
    }
```

**Review by mattdailis — CHANGES_REQUESTED** [review]

File: scheduler-worker/scheduling-dsl-compiler/src/libs/scheduler-edsl-fluent-api.ts (hunk 1 above)
Comment (jmdelfa) [reply scheduler-worker/scheduling-dsl-compiler/src/libs/scheduler-edsl-fluent-api.ts]: Updated code with spacing included

**Review by mattdailis — APPROVED** [review]

**Review by mattdailis — APPROVED** [review]

**Review by mattdailis — APPROVED** [review]

**Comment by jmdelfa** [comment]
All comments resolved. PR ready to be merged

**Comment by adrienmaillard** [comment]
WOW, triple approval, this must mean something :-) 
Screenshot 2023-08-30 at 3 20 18 PM

**Pull request merged**